game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProUtils.java

```diff
@@ -218,18 +219,46 @@ public static int getClosestEnemyLandTerritoryDistanceOverWater(final GameData d
   public static boolean isFfa(final GameData data, final PlayerID player) {
     final RelationshipTracker relationshipTracker = data.getRelationshipTracker();
     final Set<PlayerID> enemies = relationshipTracker.getEnemies(player);
-    for (final PlayerID enemy : enemies) {
-      if (relationshipTracker.isAtWarWithAnyOfThesePlayers(enemy, enemies)) {
+    final Set<PlayerID> enemiesWithoutNeutrals =
+        enemies.stream().filter(p -> !isNeutralPlayer(p)).collect(Collectors.toSet());
+    for (final PlayerID enemy : enemiesWithoutNeutrals) {
+      if (relationshipTracker.isAtWarWithAnyOfThesePlayers(enemy, enemiesWithoutNeutrals)) {
         return true;
       }
     }
     return false;
```

Comment:
As a side note: This would be an optimal use case for streams as well -> anyMatch 
You can also use `Util.not(ProUtils::isNeutralPlayer)` if you want

Reply:
Refactored to use streams.